src/MongoDB.Driver/DatabaseAggregateVoid.cs

```diff
@@ -0,0 +1,74 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using MongoDB.Bson.Serialization;
+
+namespace MongoDB.Driver
+{
+    /// <summary>
+    /// Represents the void input data type to a database aggregation pipeline. No actual instances of this type can be created.
+    /// </summary>
+    public sealed class DatabaseAggregateVoid
```

Comment:
Since our type safe PipelineDefinition requires a TInput as well as a TOutput we need *some* type to use as the TInput of a pipeline passed to the Database Aggregate method. This is that type.

Reply:
This class looks really weird, but I give up for now. I tried to find a solution at least to hide this class inside `pipeline`s. But the number of required changes for it, probably higher than we can/want do for this ticket.